### ISSUE TITLE
OSX port enhancements

### ISSUE THREAD

**Issue by jpd002** [jpd002/Play-#33]
Just opening an issue here for features that would be interesting to add in the OSX port. Here's a list of things I'm planning on adding:

- FPS counter
- Control configuration
- Proper about box.

Don't hesitate to add your own!

**Comment by Fadi5555** [comment]
@jpd002 that's nice list of features that you mentioned above.
I think it'll be nice if you add cheat selector like cwheat for psp games. because most of us like doing some cheats on games.( for example to unlock overkill weapons something or like that).
Also if you can add buffering/non buffering render (it's to make the games in full hd like on ppsspp)

This is my opinion. You really do a good job on your amazing project. Looking forward for porting a build for iOS as well.

Keep your doing up.

**Comment by kode54** [comment]
Sound support as well?

**Comment by jpd002** [comment]
Indeed, sound would be nice on OSX too. Using what's been done on the OSX PsfPlayer would probably work fine.

**Comment by Fadi5555** [comment]
@jpd002
You have done PsfPlayer long time for mac and ios.

So if support of sound is come will this includes ios? Because it's already have PsfPlayer?

**Comment by jpd002** [comment]
Yes, it would be possible to have sound on iOS, but since the frame rate is low, it would sound very bad.

**Comment by Fadi5555** [comment]
Thanks for replying. That's really nice to hear.
If have a free time. Try to support it for next weekly build because I have some 2d games and it frame rate is good. But the problem the supporting of controller is not working how ever it's connected successfully.

I have another thing? I saw you said (AOT) compilation for all platforms?
Can you tell me what is that indeed? And is it effects the black screen for most of games?

**Comment by kode54** [comment]
Ahead of Time compilation would require every supported game or PSF soundtrack to be pre-compiled into the program, which would make for rather a sizable build to support hundreds or thousands of games or soundtracks.

**Comment by Fadi5555** [comment]
@kode54 thanks for telling that. It's clear to me now what is AOT.

Well is that mean the build will change in a good way in performance of games.
Well my question: most of games crashing or staying at black screen is that will make games go further? I mean will solve the issue of them?

**Comment by kode54** [comment]
It was just an idea. To actually implement it, the developer would need to sit there and feed it every single game ever made, one at a time, to compile them into native code. Not likely to happen.

Also, Ahead of Time compilation will have no effect on compatibility, as that is controlled by other aspects of the emulator than just the emulated MIPS CPU.

**Comment by Fadi5555** [comment]
@kode54 wow that's really helpful indeed.

I'm really hope to get a high results of what we have did. I hope one day we will get compatibility looks like pcsx2 on our mobile device. Because this project is more portable.

Again thanks a lot for the information.

**Comment by Jabrwock** [comment]
Is it keyboard only on OS X? I have an XBox360-compatible controller that uses x-input.
